internal/client/ekscluster/cluster_resource.go

```diff
@@ -94,6 +101,36 @@ func (c *Client) EksClusterResourceServiceGet(fn *eksmodel.VmwareTanzuManageV1al
 	return clusterResponse, err
 }
 
+/*
+EksClusterResourceServiceGetByID gets an eks cluster by its ID.
+*/
+func (c *Client) EksClusterResourceServiceGetByID(id string) (*eksmodel.VmwareTanzuManageV1alpha1EksclusterGetEksClusterResponse, error) {
+	queryParams := url.Values{
+		"query": []string{fmt.Sprintf("uid=\"%s\"", id)},
```

Comment:
For this API, `query` is just a string. But, `url.Values` is defined as `map[string][]string`, so this is a slice of strings here.

Reply:
This is invoking a List with a cluster ID filter. Since the filter format is a bit tricky, I didn't want to expose just the List where other modules have to figure out the correct filter format.